public/scripts/upload.js

```diff
@@ -134,14 +161,38 @@ class FileListViewModel {
       this._imageUploads.push(imageUpload);
       window.viewModel.generatePreviewMockup(imageUpload);
     }
+
+    window.viewModel.selectedPreviewImageULID =
+      window.viewModel.firstImageUploadULID;
   }
 
-  async remove(filename, index) {
-    this._imageUploads = this._imageUploads.filter((upload, i) => {
+  async remove(filename, fileUlid) {
+    this._imageUploads = this._imageUploads.filter((upload) => {
       const isSameFilename = upload.file.name === filename;
-      const isSameIndex = i === index;
+      const isSameIndex = fileUlid === upload.ulid;
       return !(isSameFilename && isSameIndex);
     });
+
+    window.viewModel.selectedPreviewImageULID =
+      window.viewModel.firstImageUploadULID;
+  }
+
+  updateImageUploadStateByULID(ulid, state) {
+    this._imageUploads.forEach((imageUpload) => {
+      if (imageUpload.ulid == ulid) {
+        imageUpload.state = state;
+      }
+      return imageUpload;
```

Comment:
This return is inside `forEach()`, which IIUC have no impact.

 So 
1. it can be removed? OR
2.  do you expect `updateImageUploadStateByULID` to return the matching image?
    If so, I would work around this by using `for (const iu of this._imageUploads)`.

Reply:
update to `this._imageUploads = this._imageUploads.map(...)`